packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -0,0 +1,286 @@
+import { compare } from './private/version';
+
+/**
+ * The extra options passed to the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineBindOptions {
+  /**
+   * The role used for S3 importing.
+   *
+   * @default - none
+   */
+  readonly s3ImportRole?: iam.IRole;
+
+  /**
+   * The role used for S3 exporting.
+   *
+   *  @default - none
+   */
+  readonly s3ExportRole?: iam.IRole;
+
+  /**
+   * The customer-provided ParameterGroup.
+   *
+   * @default - none
+   */
+  readonly parameterGroup?: IParameterGroup;
+}
+
+/**
+ * The type returned from the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineBindConfig {
+  /**
+   * The ParameterGroup to use for the cluster.
+   *
+   * @default - no ParameterGroup will be used
+   */
+  readonly parameterGroup?: IParameterGroup;
+
+  /**
+   * The port to use for this cluster,
+   * unless the customer specified the port directly.
+   *
+   * @default - use the default port for clusters (3306)
+   */
+  readonly port?: number;
+}
+
+/**
+ * The interface representing a database cluster (as opposed to instance) engine.
+ */
+export interface IClusterEngine {
+  /** The type of the engine, for example "aurora-mysql". */
+  readonly engineType: string;
+
+  /**
+   * The exact version of a given engine.
+   *
+   * @default - default version for the given engine type
+   */
+  readonly engineVersion?: string;
+
+  /**
+   * The family to use for ParameterGroups using this engine.
+   * This is usually equal to "<engineType><engineMajorVersion>",
+   * but can sometimes be a variation of that.
+   * You can pass this property when creating new ClusterParameterGroup.
+   */
+  readonly parameterGroupFamily: string;
+
+  /** The application used by this engine to perform rotation for a single-user scenario. */
+  readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The application used by this engine to perform rotation for a multi-user scenario. */
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The default EC2 instance type to use for databases in this cluster. */
+  readonly defaultInstanceType: ec2.InstanceType;
+
+  /**
+   * Returns a new instance of this engine with a particular version specified.
+   * Calling {@link engineVersion} on the returned instance will always return the provided value.
+   *
+   * @param engineVersion the exact engine version, for example "5.7.mysql_aurora.2.03.4"
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-engineversion
+   */
+  withVersion(engineVersion: string): IClusterEngine;
+
+  /**
+   * Method called when the engine is used to create a new cluster.
+   */
+  bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineBindConfig;
```

Comment:
Last time I did this I was met with "why not just call it `bind()`". I guess that makes sense, so let me forward this right on to you :).

Reply:
There are actually reasons for not calling it `bind`.

1. There is a non-zero chance that there will be an engine in the future that can be used both for Instances and Clusters. If both interfaces have a `bind` method, that will make it impossible to implement both interfaces in one instance.
2. I want to avoid TypeScript's structural typing allowing you to pass an `IClusterEngine` where an `IInstanceEngine` is required, and vice versa.